--- dahdi-tools-3.1.0/xpp/mpptalk.c~	2019-10-03 16:48:11.000000000 +0200
+++ dahdi-tools-3.1.0/xpp/mpptalk.c	2020-08-24 11:51:44.540425525 +0200
@@ -237,7 +237,7 @@
 		MEMBER(MPP, TWS_PORT_GET_REPLY);
 		MEMBER(MPP, TWS_PWR_GET);
 		MEMBER(MPP, TWS_PWR_GET_REPLY);
-} PACKED members;
+} PACKED mpp_members;
 
 /*
  * Statuses
--- dahdi-tools-3.1.0/xpp/xtalk/xtalk_sync.c~	2019-10-03 16:48:11.000000000 +0200
+++ dahdi-tools-3.1.0/xpp/xtalk/xtalk_sync.c	2020-08-24 11:52:00.910944115 +0200
@@ -60,7 +60,7 @@
 	MEMBER(XTALK, ACK);
 	MEMBER(XTALK, PROTO_GET);
 	MEMBER(XTALK, PROTO_GET_REPLY);
-} PACKED members;
+} PACKED xtalk_members;
 
 const struct xtalk_protocol	xtalk_sync_proto = {
 	.name	= "XTALK-SYNC",
